### PR TITLE
LPS-74881

### PR DESCRIPTION
**Pull request by ricardocousolr**
Hi @brunobasto. This fix is for a customer. The problem arises in the navigation, within the site Liferay, when going back using the browswer's back button to a url of the form 'localhost:8080/en/colorful', where '/colorful' is the friendly url of a site page. The portal shows the loading bar indefinitely and the page never loads. This only occurs for pages starting with the letter 'c': the variable url is, in this example, 'en/colorful' while themeDisplay.getPathMain() is 'en/c'. Since the latter is the starting substring of the former the if-condition is true and a false is returned. I think that adding a character like forward-slash to both strings prevents this situation and doesn't affect all other cases. Tell me if you have any questions. Thanks!

### CODE REVIEW

**Comment by brunobasto** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/foundation/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/init.es.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ var initSPA = function(callback) {
 			{
 				handler: RenderURLScreen,
 				path: function(url) {
-					if (url.indexOf(themeDisplay.getPathMain()) === 0) {
+					if ((url + "/").indexOf(themeDisplay.getPathMain() + "/") === 0) {
 						return false;
```
Comment (brunobasto) [inline modules/apps/foundation/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/init.es.js]: Hey @ricardocousolr, thanks for sending the fix! One question:

Wouldn´t 

```if (url.indexOf(themeDisplay.getPathMain() + '/') === 0) {
	return false;
}
````

Be enough?

Comment (ricardocousolr) [reply modules/apps/foundation/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/init.es.js]: I think it wouldn't work well with the case in which the variable url is just "en/c".

**Comment by ricardocousolr** [comment]
Hi @brunobasto, do you have any other questions or foresee any other issues with the solution?

**Comment by brunobasto** [comment]
Hey @ricardocousolr, no. Sorry for the delay. I just haven't had time to test I think yet. Will semente this through today.

Thanks,

**Comment by ricardocousolr** [comment]
Thanks @brunobasto !

**Comment by brunobasto** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by brunobasto** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/51918. [See changes here.](https://github.com/brunobasto/liferay-portal/compare/c94e06a96db6~1...c94e06a96db6) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**